minecraft/src/main/java/org/quiltmc/loader/impl/game/minecraft/patch/applet/AppletLauncher.java

```diff
@@ -62,9 +63,27 @@ public AppletLauncher(File instance, String username, String sessionid, String h
 		params.put("demo", Boolean.toString(demo));
 
 		try {
-			mcApplet = (Applet) QuiltLauncherBase.getLauncher()
-					.getTargetClassLoader()
-					.loadClass(GameTransformer.appletMainClass)
+			Class<?> classToInstantiate = null;
+
+			for (String className : LibClassifier.Lib.MC_CLIENT.getPaths()) {
+				try {
+					classToInstantiate = QuiltLauncherBase
+							.getClass(
+									className
+											.replace(".class", "")
+											.replace("/",".")
+							);
+					break;
+				} catch (ClassNotFoundException exception) {
+					// Continue to next index
+				}
+			}
+
+			if (classToInstantiate == null) {
+				throw new RuntimeException("Could not instantiate MinecraftApplet - cannot find class");
+			}
+
+			mcApplet = (Applet) classToInstantiate
```

Comment:
This can easily be set via the GameProvider, unless this solution is preferred by Quilt.